// tcake-main/libs/Logger.js
/*
 * to8to-webframework
 * description: 
 * version - v1.0 2015/6/12 0012
 * author carl.wu<carl.wu@corp.to8to.com>
 * http://www.to8to.com/
 * 
 */

var colors = require('colors');

function insertBefore(msg, beforemsg) {
    var _msg;
    for (var i in msg) {
        _msg = typeof msg[i] === 'object' ? JSON.stringify(msg[i]) : msg[i];
        msg[i] = beforemsg + _msg;
    }

    return msg;
}

function logBatch(logs, color) {
    for (var i in logs) {
        console.log(logs[i][color]);
    }
}

module.exports = {

    info: function() {
        var _arguments = insertBefore(arguments, "[INFO]:");
        return logBatch(_arguments, 'green');
    },

    error: function() {
       var _arguments = insertBefore(arguments, "[ERROR]:");
        return logBatch(arguments, 'red');
    },

    warn: function() {
       var _arguments = insertBefore(arguments, "[WARN]:");
        return logBatch(arguments, 'yellow');
    },

    debug: function() {
       var _arguments = insertBefore(arguments, "[DEBUG]:");
        return logBatch(arguments, 'grey');
    },

    trace: function(msg) {
       var _arguments = insertBefore(arguments, "[TRACE]:");
        return logBatch(arguments, 'cyan');
    }

};
